### ISSUE TITLE
[Bug]: Duplicate Podcast Episodes on subscription feeds

### ISSUE THREAD

**Issue by SAS-1** [advplyr/audiobookshelf#2986]
### What happened?

Hi 

I searched and I can see #2785 but it's not quite the same issue. I think this has been discussed in Discord a few times. 

I subscribe to History Hit and have a member link for the podcasts, 24 hours before an episode is published it's made available to subscribers. This is where my problem comes in. 
It will download the episode and then the next day download the same episode this time appending a GUID to the filename. 

I forgot to grab an image from ABS front end when generating the rss xml to report this but you can see a similar example in this screenshot

![image](https://github.com/advplyr/audiobookshelf/assets/47122073/2b71de72-5f50-468d-a512-79d2470d61e2)

the episode is the same but its downloaded twice. 

### What did you expect to happen?

ABS checks if the episode has already been published and doesn't download it again. 

### Steps to reproduce the issue

1. Subscribe to podcast with a member link
2. let it download member episode
3. wait for public version of the same episode to be published
4. ABS downloads this along with the original episode

### Audiobookshelf version

2.9.0

### How are you running audiobookshelf?

Docker

### What OS is your Audiobookshelf server hosted from?

Linux

### If the issue is being seen in the UI, what browsers are you seeing the problem on?

None

### Logs

```shell
Log of Member Only version:

 <itunes:title>Stalin, Churchill and Roosevelt: The Impossible Alliance that Won WWII</itunes:title>
            <pubDate>Mon, 13 May 2024 23:30:42 GMT</pubDate>
            <itunes:duration>31:28</itunes:duration>
            <enclosure url="https://sphinx.acast.com/p/acast/s/dansnowshistoryhit/e/663e401815398a00121b6b04/media.mp3?tk=REMOVED" length="45517951" type="audio/mpeg"/>
            <guid isPermaLink="false">663e401815398a00121b6b04</guid>
            <itunes:explicit>false</itunes:explicit>
            <link>https://shows.acast.com/dansnowshistoryhit/episodes/stalin-churchill-and-roosevelt-the-impossible-alliance-that-</link>
            <acast:episodeId>663e401815398a00121b6b04</acast:episodeId>
            <acast:showId>c939f8d1-c4bc-478e-8bb9-e5343f9a7ab5</acast:showId>
            <acast:episodeUrl>stalin-churchill-and-roosevelt-the-impossible-alliance-that-</acast:episodeUrl>
            <acast:settings>REMOVED</acast:settings>
            <itunes:subtitle>How a crack team of diplomats secretly travelled to Moscow to negotiate with the intractable Stalin.</itunes:subtitle>
            <itunes:episodeType>full</itunes:episodeType>
            <itunes:episode>1438</itunes:episode>
            <itunes:summary>&lt;p&gt;In the summer of 1941, Hitler invaded the Soviet Union. As the Germans drove towards Moscow, a catastrophic Soviet defeat seemed imminent - a defeat that would have made the Allied liberation of Europe virtually impossible. To keep the Allied victory in sight, Roosevelt and Churchill assembled a crack team of diplomats to secretly travel to wartime Moscow and negotiate with the intractable Stalin.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Dan is joined by Giles Milton, bestselling historian and author of &apos;The Stalin Affair: The Impossible Alliance That Won the War&apos;. Giles tells us how the leaders who had wanted to destroy Stalin&apos;s Russia ended up desperately trying to keep it afloat.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Produced by James Hickmann and edited by Dougal Patmore.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Enjoy unlimited access to award-winning original documentaries that are released weekly and AD-FREE podcasts. Get a subscription for £1 per month for 3 months with code &lt;strong&gt;DANSNOW&lt;/strong&gt; - sign up at &lt;a href=&quot;https://historyhit.com/subscription/&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt;https://historyhit.com/subscription/&lt;/a&gt;.&lt;/p&gt;&lt;br&gt;&lt;p&gt;We&apos;d love to hear from you - what do you want to hear an episode on? You can email the podcast at &lt;a href=&quot;mailto:ds.hh@historyhit.com&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt;ds.hh@historyhit.com&lt;/a&gt;.&lt;/p&gt;&lt;br&gt;&lt;p&gt;You can take part in our listener survey&lt;a href=&quot;https://www.surveymonkey.co.uk/r/6FFT7MK&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt; here&lt;/a&gt;.&lt;/p&gt;</itunes:summary>
            <title>Stalin, Churchill and Roosevelt: The Impossible Alliance that Won WWII</title>
            <description>&lt;p&gt;In the summer of 1941, Hitler invaded the Soviet Union. As the Germans drove towards Moscow, a catastrophic Soviet defeat seemed imminent - a defeat that would have made the Allied liberation of Europe virtually impossible. To keep the Allied victory in sight, Roosevelt and Churchill assembled a crack team of diplomats to secretly travel to wartime Moscow and negotiate with the intractable Stalin.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Dan is joined by Giles Milton, bestselling historian and author of &apos;The Stalin Affair: The Impossible Alliance That Won the War&apos;. Giles tells us how the leaders who had wanted to destroy Stalin&apos;s Russia ended up desperately trying to keep it afloat.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Produced by James Hickmann and edited by Dougal Patmore.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Enjoy unlimited access to award-winning original documentaries that are released weekly and AD-FREE podcasts. Get a subscription for £1 per month for 3 months with code &lt;strong&gt;DANSNOW&lt;/strong&gt; - sign up at &lt;a href=&quot;https://historyhit.com/subscription/&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt;https://historyhit.com/subscription/&lt;/a&gt;.&lt;/p&gt;&lt;br&gt;&lt;p&gt;We&apos;d love to hear from you - what do you want to hear an episode on? You can email the podcast at &lt;a href=&quot;mailto:ds.hh@historyhit.com&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt;ds.hh@historyhit.com&lt;/a&gt;.&lt;/p&gt;&lt;br&gt;&lt;p&gt;You can take part in our listener survey&lt;a href=&quot;https://www.surveymonkey.co.uk/r/6FFT7MK&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt; here&lt;/a&gt;.&lt;/p&gt;</description>
            <podaccess:premium locked="false"/>
            <itunes:image href="https://assets.pippa.io/shows/6215f67d4b795a40fffd3b2c/1667286484069-7fc0af1db619130dc8ff794c866b6fa8.jpeg"/>

Log of Public version published 24 hours later:

<title>Stalin, Churchill and Roosevelt: The Impossible Alliance that Won WWII</title>
            <itunes:title>Stalin, Churchill and Roosevelt: The Impossible Alliance that Won WWII</itunes:title>
            <pubDate>Tue, 14 May 2024 23:30:59 GMT</pubDate>
            <itunes:duration>31:28</itunes:duration>
            <enclosure url="https://sphinx.acast.com/p/acast/s/dansnowshistoryhit/e/663e401815398a00121b6b04/media.mp3?tk=REMOVED" length="45517951" type="audio/mpeg"/>
            <guid isPermaLink="false">663e401815398a00121b6b04</guid>
            <itunes:explicit>false</itunes:explicit>
            <link>https://shows.acast.com/dansnowshistoryhit/episodes/stalin-churchill-and-roosevelt-the-impossible-alliance-that-</link>
            <acast:episodeId>663e401815398a00121b6b04</acast:episodeId>
            <acast:showId>c939f8d1-c4bc-478e-8bb9-e5343f9a7ab5</acast:showId>
            <acast:episodeUrl>stalin-churchill-and-roosevelt-the-impossible-alliance-that-</acast:episodeUrl>
            <acast:settings>REMOVED</acast:settings>
            <itunes:subtitle>How a crack team of diplomats secretly travelled to Moscow to negotiate with the intractable Stalin.</itunes:subtitle>
            <itunes:episodeType>full</itunes:episodeType>
            <itunes:episode>1438</itunes:episode>
            <description>&lt;p&gt;In the summer of 1941, Hitler invaded the Soviet Union. As the Germans drove towards Moscow, a catastrophic Soviet defeat seemed imminent - a defeat that would have made the Allied liberation of Europe virtually impossible. To keep the Allied victory in sight, Roosevelt and Churchill assembled a crack team of diplomats to secretly travel to wartime Moscow and negotiate with the intractable Stalin.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Dan is joined by Giles Milton, bestselling historian and author of &apos;The Stalin Affair: The Impossible Alliance That Won the War&apos;. Giles tells us how the leaders who had wanted to destroy Stalin&apos;s Russia ended up desperately trying to keep it afloat.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Produced by James Hickmann and edited by Dougal Patmore.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Enjoy unlimited access to award-winning original documentaries that are released weekly and AD-FREE podcasts. Get a subscription for £1 per month for 3 months with code &lt;strong&gt;DANSNOW&lt;/strong&gt; - sign up at &lt;a href=&quot;https://historyhit.com/subscription/&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt;https://historyhit.com/subscription/&lt;/a&gt;.&lt;/p&gt;&lt;br&gt;&lt;p&gt;We&apos;d love to hear from you - what do you want to hear an episode on? You can email the podcast at &lt;a href=&quot;mailto:ds.hh@historyhit.com&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt;ds.hh@historyhit.com&lt;/a&gt;.&lt;/p&gt;&lt;br&gt;&lt;p&gt;You can take part in our listener survey&lt;a href=&quot;https://www.surveymonkey.co.uk/r/6FFT7MK&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt; here&lt;/a&gt;.&lt;/p&gt;</description>
            <itunes:summary>&lt;p&gt;In the summer of 1941, Hitler invaded the Soviet Union. As the Germans drove towards Moscow, a catastrophic Soviet defeat seemed imminent - a defeat that would have made the Allied liberation of Europe virtually impossible. To keep the Allied victory in sight, Roosevelt and Churchill assembled a crack team of diplomats to secretly travel to wartime Moscow and negotiate with the intractable Stalin.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Dan is joined by Giles Milton, bestselling historian and author of &apos;The Stalin Affair: The Impossible Alliance That Won the War&apos;. Giles tells us how the leaders who had wanted to destroy Stalin&apos;s Russia ended up desperately trying to keep it afloat.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Produced by James Hickmann and edited by Dougal Patmore.&lt;/p&gt;&lt;br&gt;&lt;p&gt;Enjoy unlimited access to award-winning original documentaries that are released weekly and AD-FREE podcasts. Get a subscription for £1 per month for 3 months with code &lt;strong&gt;DANSNOW&lt;/strong&gt; - sign up at &lt;a href=&quot;https://historyhit.com/subscription/&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt;https://historyhit.com/subscription/&lt;/a&gt;.&lt;/p&gt;&lt;br&gt;&lt;p&gt;We&apos;d love to hear from you - what do you want to hear an episode on? You can email the podcast at &lt;a href=&quot;mailto:ds.hh@historyhit.com&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt;ds.hh@historyhit.com&lt;/a&gt;.&lt;/p&gt;&lt;br&gt;&lt;p&gt;You can take part in our listener survey&lt;a href=&quot;https://www.surveymonkey.co.uk/r/6FFT7MK&quot; rel=&quot;noopener noreferrer&quot; target=&quot;_blank&quot;&gt; here&lt;/a&gt;.&lt;/p&gt;</itunes:summary>
            <itunes:image href="https://assets.pippa.io/shows/6215f67d4b795a40fffd3b2c/1667286484069-7fc0af1db619130dc8ff794c866b6fa8.jpeg"/>
```

### Additional Notes

I have removed my unique keys I can share the link via private message if needed for testing. 

I think what ABS needs is a check episodeId, episode and title is the same don't download as new.

**Comment by advplyr** [comment]
Abs uses the GUID as the unique identifier for episodes as per specs. Although, I've never seen an RSS feed use the same GUID twice in the feed, I think this violates RSS feed specs. Just to clarify, you can see in the RSS feed that the episode is listed twice with the same GUID?

**Comment by SAS-1** [comment]
Hi

So day 1 the episode is published in the rss feed, then day 2 it updates/replaces the members version with the public one.

Same file, same details, just the date seems to change.

It happens with all their feeds, I did report this to them last year but they haven't a clue how it works according to them they give acast the files with a release schedule and 24 hours early access for subscribers.

It's not a major issue just annoying as I only keep 3 episodes and 2 are always the same with this issue.

**Comment by nichwall** [comment]
Yeah, the GID is specified as "unique".
https://www.rssboard.org/rss-specification#ltguidgtSubelementOfLtitemgt

**Comment by advplyr** [comment]
Have you looked at the full RSS feed to check if the episode is listed twice? You are saying it is replacing it in the feed but I'm not sure if you are looking at the full RSS feed or not

**Comment by SAS-1** [comment]
Yeah i did a search in the xml, could onlysee it once, I only extracted the example one but I can redo the query in postman when the next episode drops and then 24 hours later and send you it.

**Comment by advplyr** [comment]
I found a potential issue that this could happen if the enclosure URL was changing between the early episode drop and the one 24 hours later. I'm pretty sure this will fix it for the next release, or you can use the `edge` docker image if you want to test it before the next server release.

**Comment by SAS-1** [comment]
cool thanks, I am on edge so I will pull it and see when they next release an episode. looking at it I expect it is the enclosure url it looks like the tk and sig params are slightly different.

**Comment by SAS-1** [comment]
That seems to have fixed it, one of the feeds updated the same episode overnight as the day before and it never downloaded it:

![image](https://github.com/advplyr/audiobookshelf/assets/47122073/9bc86171-d3f1-4ece-818a-d1792349e074)
![image](https://github.com/advplyr/audiobookshelf/assets/47122073/84737e62-6afd-4519-be14-71a49dc69111)

Thanks for fixing it so quickly it's been bugging me for ages :D

**Comment by advplyr** [comment]
Fixed in [v2.10.0](https://github.com/advplyr/audiobookshelf/releases/tag/v2.10.0)
